test/test_feature_set.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+from pytest import mark, param
+
+from lhotse.audio import AudioSet
+from lhotse.features import FeatureSet, FeatureExtractor, FeatureSegment
+from lhotse.supervision import SupervisionSet
+
+other_params = {}
+some_augmentation = None
+
+
+def test_create_feature_set():
+    # TODO(pzelasko): will be split into several tests
+
+    # Use case #1 - pre-computed features, stored somewhere, specified by an existing manifest
+    feature_set = FeatureSet.from_yaml('...')
+
+    # Use case #2
+    # - feature extraction
+    # - with recordings delivered by AudioSet,
+    # - with specified feature parameters (mfcc/fbank/..., frame shapes, etc.),
+    # - with specified optional segmentation (default: extract feats for whole recordings)
+    # Note: introduce FeatureExtractor as a FeatureSet builder for separation of concerns (loading and manipulating feauters vs extracting them)
+
+    audio_set = AudioSet.from_yaml('...')
+
+    # Variant A: whole recordings
+    whole_recording_feature_set: FeatureSet = (
+        FeatureExtractor()
+            .with_audio_set(audio_set)
+            .with_augmentation(some_augmentation)
+            .with_algorithm(method='mfcc', frame_size=0.025, frame_shift=0.01, **other_params)
+            .extract()
+    )
+
+    # Variant B: Exact segments from SupervisionSet
+    supervision_set = SupervisionSet.from_yaml('...')
+
+    whole_recording_feature_set: FeatureSet = (
+        FeatureExtractor()
+            .with_audio_set(audio_set)
+            .with_augmentation(some_augmentation)
+            .with_algorithm(method='mfcc', frame_size=0.025, frame_shift=0.01, **other_params)
+            .with_segmentation(supervision_set, extra_left_seconds=0.0, extra_right_seconds=0.0)
+            .extract()
+    )
+
+    # Variant C: custom segmentation
+    segmentation = [
+        #               recording-id  channel  start  duration
+        FeatureSegment('recording-1', 0, 0.5, 0.5),
+        FeatureSegment('recording-1', 1, 1.3, 1.5),
+        FeatureSegment('recording-1', 0, 2.4, 1.1),
+        FeatureSegment('recording-2', 0, 0.0, 30.)
+    ]
+    segment_feature_set: FeatureSet = (
+        FeatureExtractor()
+            .with_audio_set(audio_set)
+            .with_augmentation(some_augmentation)
+            .with_algorithm(method='mfcc', frame_size=0.025, frame_shift=0.01, **other_params)
+            .with_segmentation(segmentation)
+            .extract()
+    )
+
+
+@mark.parametrize(
```

Comment:
this confuses me (and likely others)... please explain in comment.

Reply:
that's a `pytest` utility, which allows you to use the same test with different arguments (parametrized tests); simple example:

```
@pytest.mark.parametrize(
  ['a', 'b', 'c'],
  [(1, 1, 2), (1, 1, 3)]
)
def test_addition(a, b, c):
  assert a + b == c
```

the first argument to parametrize describes the names of inputs (`['a', 'b', 'c']`, and the second argument is a list of tuples or arguments (`[(1, 1, 2), (1, 1, 3)]`)

the second case would fail, as you might expect it to, so `pytest` allows you to mark it as "expected fail", in short, `xfail`, hence we should instead write `param(1, 1, 3, mark=pytest.mark.xfail)`